kvbc/test/v4blockchain/v4_blockchain_test.cpp

```diff
@@ -45,6 +48,106 @@ class v4_kvbc : public Test {
     ASSERT_EQ(0, db.use_count());
     cleanup();
   }
+  void create_blocks(uint64_t num_blocks,
+                     uint32_t& num_merkle_each,
+                     uint32_t& num_versioned_each,
+                     uint32_t& num_immutable_each) {
+    std::map<std::string, categorization::CATEGORY_TYPE> cat_map{
+        {"merkle", categorization::CATEGORY_TYPE::block_merkle},
+        {"versioned", categorization::CATEGORY_TYPE::versioned_kv},
+        {"immutable", categorization::CATEGORY_TYPE::immutable}};
+    v4blockchain::KeyValueBlockchain blockchain{db, true, cat_map};
+    std::mt19937 rgen;
+    std::uniform_int_distribution<uint32_t> dist(10, 100);
+    num_merkle_each = dist(rgen);
+    num_versioned_each = dist(rgen);
+    num_immutable_each = dist(rgen);
+    // Keys are:
+    // <category_name>_key_<blockId>_<key_id>
+    // Values are:
+    // <category_name>_value_<blockId>_<key_id>
+    for (uint64_t blk = 1; blk <= num_blocks; ++blk) {
+      categorization::Updates updates;
+
+      categorization::BlockMerkleUpdates merkle_updates;
+      for (uint32_t kid = 1; kid <= num_merkle_each; ++kid) {
+        std::string key = "merkle_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+        std::string val = "merkle_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+        merkle_updates.addUpdate(std::move(key), std::move(val));
+      }
+      updates.add("merkle", std::move(merkle_updates));
+
+      categorization::VersionedUpdates ver_updates;
+      ver_updates.calculateRootHash(false);
+      for (uint32_t kid = 1; kid <= num_versioned_each; ++kid) {
+        std::string key = "versioned_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+        std::string val = "versioned_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+        ver_updates.addUpdate(std::move(key), std::move(val));
+      }
+      updates.add("versioned", std::move(ver_updates));
+
+      categorization::ImmutableUpdates immutable_updates;
+      for (uint32_t kid = 1; kid <= num_immutable_each; ++kid) {
+        std::string key = "immutable_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+        std::string val = "immutable_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+        immutable_updates.addUpdate(std::move(key), {std::move(val), {std::to_string(blk), std::to_string(kid)}});
+      }
+      updates.add("immutable", std::move(immutable_updates));
+      ASSERT_EQ(blockchain.add(std::move(updates)), (BlockId)blk);
+    }
+  }
+
+  void delete_blocks(uint64_t& tot_num_blocks,
```

Comment:
the name should be different, like add deletes to block

Reply:
Thanks @cloudnoize updating the name to add_deletes_to_blocks()